frontend/src/scenes/data-warehouse/editor/sidebar/graph/UpstreamGraph.tsx

```diff
@@ -210,6 +208,27 @@ function UpstreamGraphContent({ codeEditorKey }: UpstreamGraphProps): JSX.Elemen
 
     const nodeTypes = useMemo(() => getNodeTypes(edges), [edges])
 
+    const coloredEdges: Edge[] = edges.map((edge) => {
```

Comment:
Maybe consider using a `memo` here - im not so sure how often this gets rerendered though, so your call